libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Logger/SFLogger.m

```diff
@@ -272,7 +364,13 @@ + (nonnull instancetype)loggerForComponent:(nonnull NSString *)componentName {
         }
         id logger = [loggerList objectForKey:componentName];
         if (!logger) {
-            logger =  [[self alloc] init:componentName];
+            id<SFLogReceiver> logReceiver = nil;
+            if (_logReceiverFactory) {
+                logReceiver = [_logReceiverFactory createWithComponentName:componentName];
```

Comment:
This is where the new factory injects the log receiver.  Much like for Android, I'm hoping an app could implement their dependency injection in their factory if desired.